### PR TITLE
README improvements

### PR DESCRIPTION
**Pull request by gpaul**
This PR makes various improvements to the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,27 +2,179 @@
...
+
+The `./pkg/csilvm/server.go` file should serve as your entrypoint when reading the code as it includes all the CSI RPC endpoints.
 
-Logical volumes are named according to the following pattern
-`<volume-group-name>_<logical-volume-name>`.
 
-## Runtime dependencies
+## Deployment
+
+The plugin builds to a single executable binary called `csilvm`.
```
Comment (jieyu) [inline README.md]: Does the binary has any dependency, or is it self contained (i.e., static)?

File: README.md [hunk 2]
```diff
@@ -33,32 +185,73 @@ The following command-line utilties must be present in the `PATH`:
...
 This project uses JIRA instead of GitHub issues to track bugs and feature requests.
 You may review the [currently open issues](https://jira.mesosphere.com/issues/?jql=project%20%3D%20DCOS_OSS%20and%20component%20%3D%20csilvm%20and%20status%20not%20in%20(Resolved%2C%20Closed)).
 You may also create a new [bug](https://jira.mesosphere.com/secure/CreateIssueDetails!init.jspa?pid=14105&issuetype=1&components=20732&customfield_12300=3&summary=CSILVM%3a+bug+summary+goes+here&description=Environment%3a%0d%0dWhat+you+attempted+to+do%3a%0d%0dThe+result+you+expected%3a%0d%0dThe+result+you+saw+instead%3a%0d&priority=3) or a new [task](https://jira.mesosphere.com/secure/CreateIssueDetails!init.jspa?pid=14105&issuetype=3&components=20732&customfield_12300=3&summary=CSILVM%3a+task+summary+goes+here&priority=3).
+
+
+# Authors
+
+* @gpaul
+* @jdef
+* @jieyi
```
Comment (jieyu) [inline README.md]: @jieyu

File: README.md [hunk 3]
```diff
@@ -33,32 +185,73 @@ The following command-line utilties must be present in the `PATH`:
...
+
+
+# License
+
+This project is licensed under the Apache 2.0 License - see the LICENSE fail for details.
```
Comment (jieyu) [inline README.md]: typo: file

File: README.md (hunk 1 above)
Comment (gpaul) [reply README.md]: It is _not_ static in that it doesn't use a pure-go networking stack and instead links dynamically against libc for networking. Specifically,

```
$ ldd ./csilvm
	linux-vdso.so.1 (0x00007fffc96a7000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f98146cb000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f9814315000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f98148e9000)
```

If we use a pure-go networking stack one could generate a static binary:
```
$ go build -a -tags netgo ./cmd/csilvm
$ ldd ./csilvm
	not a dynamic executable
```

We can add a Make target for building a static binary.

File: README.md [hunk 4]
```diff
@@ -2,27 +2,179 @@
...
 
-Logical volumes are named according to the following pattern
-`<volume-group-name>_<logical-volume-name>`.
 
-## Runtime dependencies
+## Deployment
+
+The plugin builds to a single executable binary called `csilvm`.
+This binary should be copied to the node.
+It is expected that the CO will launch the binary using the appropriate command-line flags.
```
Comment (jdef) [inline README.md]: s/CO/Plugin Supervisor/

**Pull request merged**